gitlab/resource_gitlab_instance_cluster_test.go

```diff
@@ -0,0 +1,275 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabInstanceClusterExists("gitlab_instance_cluster.foo", &cluster),
+					testAccCheckGitlabInstanceClusterAttributes(&cluster, &testAccGitlabInstanceClusterExpectedAttributes{
+						Name:                        fmt.Sprintf("foo-cluster-%d", rInt),
+						Domain:                      "example.com",
+						EnvironmentScope:            "*",
+						KubernetesApiURL:            "https://123.123.123",
+						KubernetesCACert:            instanceClusterFakeCert,
+						KubernetesAuthorizationType: "abac",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGitlabInstanceCluster_import(t *testing.T) {
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabInstanceClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabInstanceClusterConfig(rInt, true),
+			},
+			{
+				ResourceName:            "gitlab_instance_cluster.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"enabled", "kubernetes_token", "managed"},
+			},
+		},
+	})
+}
+
+type testAccGitlabInstanceClusterExpectedAttributes struct {
+	Name                        string
+	Domain                      string
+	EnvironmentScope            string
+	KubernetesApiURL            string
+	KubernetesCACert            string
+	KubernetesAuthorizationType string
+	ManagementProjectID         string
+	Managed                     bool
+}
+
+func testAccCheckGitlabInstanceClusterExists(n string, cluster *gitlab.InstanceCluster) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("not found: %q", n)
+		}
+
+		instanceClusterID, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		conn := testAccProvider.Meta().(*gitlab.Client)
+
+		gotCluster, _, err := conn.InstanceCluster.GetCluster(instanceClusterID)
+		if err != nil {
+			return err
+		}
+
+		*cluster = *gotCluster
+
+		return nil
+	}
+}
+
+func testAccCheckGitlabInstanceClusterDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*gitlab.Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "gitlab_instance_cluster" {
+			continue
+		}
+
+		instanceClusterID, err := strconv.Atoi(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+
+		gotCluster, resp, err := conn.InstanceCluster.GetCluster(instanceClusterID)
+
+		if err == nil {
+			if gotCluster != nil && gotCluster.ID == instanceClusterID {
+				return fmt.Errorf("instance cluster still exists")
+			}
+		}
+		if resp.StatusCode != 404 {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckGitlabInstanceClusterAttributes(cluster *gitlab.InstanceCluster, want *testAccGitlabInstanceClusterExpectedAttributes) resource.TestCheckFunc {
```

Comment:
Can you add quality checks for `ManagementProjectID` and `Managed` to this function?

Reply:
`Managed` wasn't an instance cluster attribute, adding it here: https://github.com/xanzy/go-gitlab/pull/940

Reply:
Ok. I don't think it needs to block this PR as long as we create a new issue to follow up on it. I'm not sure how soon go-gitlab will release, as I'm not familiar with that project.